code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_core.py

```diff
@@ -0,0 +1,290 @@
+        """Returns True if the experiment is valid, which means the electric range of the profile
+        is between the limits of the battery.
+        
+        If the profile has no current neither voltage, means the profile could be only of waits, so
+        is also valid.
+        If the profile has no current or voltage, means the profile could have only current/voltage
+        modes apart from waits, so is also valid if the range of the profile is inside the range of
+        battery
+
+        Args:
+            battery (CyclerDataBatteryC): [description]
+            profile (CyclerDataProfileC): [description]
+
+        Returns:
+            bool: [description]
+        """
+        # Check if the profile range is inside the battery range
+        res = False
+        if profile.range.no_voltage() and profile.range.no_current():
+            log.debug("Profile no current neither voltages")
+            res = True
+        elif profile.range.no_current():
+            log.debug("Profile no current")
+            res = profile.range.in_range_voltage(battery.elec_ranges)
+        elif profile.range.no_voltage():
+            log.debug("Profile no voltage")
+            res = profile.range.in_range_current(battery.elec_ranges)
+        else:
+            log.debug("Profile with current and voltage")
+            res = (profile.range.in_range_current(battery.elec_ranges) and
+                profile.range.in_range_voltage(battery.elec_ranges))
+        log.critical(f"Validating experiment: {res}")
+        return res
+
+    def __update_exp_status(self, exp_status: CyclerDataExpStatusE) -> None:
+        """Write the experiment status in the shared memory
+
+        Args:
+            exp_status (CyclerDataExpStatusE): Experiment status
+        """
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.SET_EXP_STATUS,
+                        exp_status= exp_status)
+        self.__chan_str_reqs.send_data(request)
+
+    def turn_deprecated(self):
+        """Send a deprecated command to the str node to turn all experiments to error
+        """
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.TURN_DEPRECATED)
+        self.__chan_str_reqs.send_data(request)
+
+    def __execute_experiment(self) -> None:
+        """Execute the experiment
+        """
+        # First step is to update the local data in power
+        self.exp_status, self.__local_gen_meas.instr_id = self.pwr_control.process_iteration()
+        self.__update_exp_status(self.exp_status)
+
+    def execute_machine_status(self) -> None: #pylint: disable=too-many-branches
+        """Execute the machine status
+        """
+        log.debug("Executing machine status")
+        try:
+            ## Process machine status
+            if self.state == AppManCoreStatusE.GET_EXP:
+                ## Wait for cs status to continue
+                if ((not self.__wait_cs_reqst and not self.__wait_exp_reqst) or
```

Comment:
I think is not a good idea to have wait_cs and wait_exp

Reply:
It can cause to send more than one request?

Reply:
Disscuss in future versions